Docs add lonboard tutorial (#529)
microsoft/PlanetaryComputerDataCatalog: docs/overview/lonboard.md

# Visualizing Planetary Computer data with Lonboard

> [Lonboard](https://developmentseed.org/lonboard/) is a Python library for interactive geospatial visualization in Jupyter. It renders large vector datasets on a GPU-accelerated WebGL map directly in the notebook, with no tile server in the loop. Geometry streams to the browser as [Apache Arrow](https://arrow.apache.org/), so hundreds of thousands of features stay interactive. Layers compose: stack footprints, points, and analysis results in a single map.

Reviewer: ```suggestion
[Lonboard](https://developmentseed.org/lonboard/) is a Python library for interactive geospatial visualization in Jupyter. It renders large vector datasets on a GPU-accelerated WebGL map directly in the notebook, with no tile server in the loop. Geometries stream to the browser as [Apache Arrow](https://arrow.apache.org/), so hundreds of thousands of features stay interactive. Layers compose: stack footprints, points, and analysis results in a single map.
```

## Connect to the Planetary Computer STAC catalog

Set up the catalog client with PC's signer so every search result has a signed asset href:

```python
import pystac_client
import planetary_computer

catalog = pystac_client.Client.open(
    "https://planetarycomputer.microsoft.com/api/stac/v1",
    modifier=planetary_computer.sign_inplace,
)
```

`modifier=planetary_computer.sign_inplace` signs every asset as the search returns.

## Find the building-footprints partition for Portland

We'll render [Microsoft Building Footprints](https://planetarycomputer.microsoft.com/dataset/ms-buildings), a dataset partitioned by [quadkey](https://learn.microsoft.com/en-us/bingmaps/articles/bing-maps-tile-system). Compute the zoom-9 quadkey for a Portland coordinate and fetch the STAC item whose partition covers it:

```python
import math

def quadkey(lat, lon, zoom):
    n = 2 ** zoom
    x = int((lon + 180.0) / 360.0 * n)
    y = int((1.0 - math.asinh(math.tan(math.radians(lat))) / math.pi) / 2.0 * n)
    digits = []
    for i in range(zoom, 0, -1):
        bit = 1 << (i - 1)
        digits.append(str((1 if x & bit else 0) + (2 if y & bit else 0)))

>     return "".join(digits)

Reviewer: This is pretty ugly and not user friendly. If I saw this as an intro user I'd be scared away. 

Even though it hasn't been updated in a while, I think [Mercantile](https://github.com/mapbox/mercantile) is still the gold standard here. 

You can use [`mercantile.tile`](https://mercantile.readthedocs.io/en/stable/api/mercantile.html#mercantile.tile) to convert a lon/lat pair to a web mercator tile and then use [`quadkey`](https://mercantile.readthedocs.io/en/stable/api/mercantile.html#mercantile.quadkey) to convert to a string. 

Also you don't want to cast a quadkey to an `int`. A quadkey can start with a leading `0` and be valid.

> qk = quadkey(45.52, -122.66, 9)

Reviewer: As mentioned above
```suggestion
import mercantile

tile = mercantile.tile(-122.66, 45.52, 9)
quadkey = mercantile.quadkey(*tile)
```

item = next(catalog.search(
    collections=["ms-buildings"],
    query={
        "msbuildings:region": {"eq": "UnitedStates"},

>         "msbuildings:quadkey": {"eq": int(qk)},

Reviewer: ```suggestion
        "msbuildings:quadkey": {"eq": quadkey},
```

    },
).items())
asset = item.assets["data"]
```

## Load the footprints into a GeoDataFrame

The asset is a Delta/Parquet partition on Azure Blob. `dask_geopandas.read_parquet` reads it with the asset's `table:storage_options` (account + SAS), then `.compute()` materializes a GeoDataFrame. Clip to the Portland metro for a focused view:

```python
import dask_geopandas

gdf = dask_geopandas.read_parquet(
    asset.href,
    storage_options=asset.extra_fields["table:storage_options"],
).compute()
gdf = gdf.cx[-122.85:-122.45, 45.42:45.62]

len(gdf)  # a few hundred thousand buildings

> ```

Reviewer: I'd recommend following the behavior of the [existing notebook](https://planetarycomputer.microsoft.com/dataset/ms-buildings#Example-Notebook) to load this data. We don't need `dask_geopandas` (dask is unnecessary for the example) but we should use `deltalake` to load the data

## Render the footprints

> `PolygonLayer.from_geopandas()` uploads the geometry to the GPU as Arrow. Drawing the footprints as outlines keeps every building legible at city scale, and the map stays fully interactive with no tile server in the loop:

Reviewer: ```suggestion
[`PolygonLayer.from_geopandas()`](https://developmentseed.org/lonboard/latest/api/layers/polygon-layer/#lonboard.PolygonLayer.from_geopandas) uploads the geometry to the GPU as Arrow. Drawing the footprints as outlines keeps every building legible at city scale, and the map stays fully interactive with no tile server in the loop:
```

# Visualizing Planetary Computer data with Lonboard

[Lonboard](https://developmentseed.org/lonboard/) is a Python library for interactive geospatial visualization in Jupyter. It renders large vector datasets on a GPU-accelerated WebGL map directly in the notebook, with no tile server in the loop. Geometries stream to the browser as [Apache Arrow](https://arrow.apache.org/), so hundreds of thousands of features stay interactive. Layers compose: stack footprints, points, and analysis results in a single map.

A companion notebook walks through every step end-to-end with live maps. [Open in Planetary Computer Hub](https://pccompute.westeurope.cloudapp.azure.com/compute/hub/user-redirect/git-pull?repo=https://github.com/microsoft/PlanetaryComputerExamples&urlpath=lab/tree/PlanetaryComputerExamples/quickstarts/lonboard.ipynb&branch=main)

## Install Lonboard

```bash
uv add lonboard pystac-client planetary-computer geopandas deltalake adlfs mercantile
```

> `pystac-client` queries the Planetary Computer STAC API; `planetary-computer` signs asset URLs; `deltalake` opens the Delta Table partition for the dataset; `geopandas` + `adlfs` read the parquet files straight off Azure Blob; `mercantile` converts a longitude/latitude to a quadkey for the search query.

Reviewer: Nit: I think `Azure Blob storage` reads better than just `Azure Blob`
